catalogue/views.py

```diff
@@ -30,6 +32,17 @@ class SearchView(FilterView):
     queryset = Component.public_objects.all()
     filterset_class = ComponentFilter
 
+    def get(self, request, *args, **kwargs):
+        # check if user has session and save query
+        if not request.session or not request.session.session_key:
+            request.session.save()
+        log = SearchLog.objects.create(
+            query=request.get_full_path(), identifier=request.session.session_key
+        )
+        # Save SearchLog in session for component relation
+        request.session["current_query"] = log.id
```

Comment:
that way, some visits could be falsely connected to a previous search. not perfect but should be fine for now.